product-scenarios/3-securing-an-api-with-different-security-protocols/3.1-api-developers-want-to-secure-their-api-using-an-authentication-machanism/3.1.1-api-developers-and-publishers-want-to-make-their-apis-secure-using-the-oauth2-protocol/src/test/java/org/wso2/am/scenario/tests/api/secure/oauth2/SecureUsingAuth2TestCase.java

```diff
@@ -120,10 +127,87 @@ public void testOAuth2Authorization(String tokenPrefix) throws Exception {
+                requestHeaders);
+        assertEquals(apiResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "Response code mismatched when api invocation. \n API response : " + apiResponse.getData());
+    }
+
+    @Test(description = "3.1.1.4")
+    public void testResourceApplicationInvokeByCustomAuthorization() throws Exception {
+        changeCustomAuthorizationHeaderInAPI(CUSTOM_AUTH_HEADER);
+        Map<String, String> requestHeaders = new HashMap();
+        requestHeaders.put(CUSTOM_AUTH_HEADER, "Bearer " + accessToken);
+        requestHeaders.put("Content-Type", "application/x-www-form-urlencoded");
+        String gatewayHttpsUrl = getHttpsAPIInvocationURL(TEST_API_1_CONTEXT, TEST_API_1_VERSION,
+                "/CheckPhoneNumber");
+        HttpResponse apiResponse = HttpClient.doGet(gatewayHttpsUrl + "?PhoneNumber=18006785432&LicenseKey=0",
+                requestHeaders);
+        assertEquals(apiResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "Response code mismatched when api invocation. \n API response : " + apiResponse.getData());
+        changeCustomAuthorizationHeaderInAPI("");
+    }
+
+    @Test(description = "3.1.1.7")
+    public void testResourceSetSecurityTypeAsNoneCanInvokedAPIWithoutTokenHeader() throws Exception {
+        Map<String, String> requestHeaders;
+        requestHeaders = new HashMap<>();
+        requestHeaders.put("Content-Type", "application/x-www-form-urlencoded");
+
+        List<NameValuePair> urlParameters = new ArrayList<>();
+        urlParameters.add(new BasicNameValuePair("PhoneNumbers", "18006785432"));
+        urlParameters.add(new BasicNameValuePair("LicenseKey", "0"));
+
+        String gatewayHttpsUrl = getHttpsAPIInvocationURL(TEST_API_1_CONTEXT, TEST_API_1_VERSION,
+                "/CheckPhoneNumbers");
+        HttpResponse apiResponse = HttpClient.doPost(gatewayHttpsUrl, requestHeaders, urlParameters);
+        assertEquals(apiResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "Response code mismatched when api invocation. \n API response : " + apiResponse.getData());
+    }
+
+    public void changeCustomAuthorizationHeaderInAPI(String customAuth) throws Exception {
+        HttpResponse getSwaggerResponse = apiPublisher.getSwagger("PhoneVerifyAPI-1", "1.0.0", API_DEVELOPER_USERNAME);
+        APIManageBean apiManageBean = new APIManageBean("PhoneVerifyAPI-1", "1.0.0", API_DEVELOPER_USERNAME,
+                "https", "disabled",
+                "resource_level", "Production and Sandbox", getSwaggerResponse.getData(),
+                "Unlimited,Gold,Bronze");
+
+        apiManageBean.setAuthorizationHeader(customAuth);
+        HttpResponse apiManageResponse = apiPublisher.manageAPI(apiManageBean);
+        verifyResponse(apiManageResponse);
+
+        org.wso2.am.integration.test.utils.bean.APILifeCycleStateRequest updateLifeCycle =
```

Comment:
Any particular reason to use the class name with the package name rather than importing the it at the top of the class?

Reply:
Fixed. There are two classes with this same name in two different places. But in here it is possible to import it in the top since we used one class within this class.